en/21/06.md

```diff
@@ -0,0 +1,199 @@
+
+ * By value, which means that the ownership over the parameter's value is passed to your function. This blocks the use of the initial parameter once the function call had ended since having it consummed by the function.
+ * By reference, which means that your function is borrowing the value of the parameter. Thus, this doesn't allow the function to change the value of the parameter since it is just `borrowed`. Once the function call finished, the ownership of the parameter returns to its initial owner.
+ * By mutable reference, which means that your function is borrowing the value of the parameter and gets the permission to change it. Once the function call finished, the ownership of the parameter returns to its initial owner keeping all changes done inside the function.
+
+```
+pub fn my function(value: u32, reference: &u32, mutable_reference: &mut u32) {
+  ...
+}
+```
+
+> Note: It's convention (but not required) to start function parameter variable names with an underscore (`_`) if the parameter is not used within the function. We'll use that convention throughout our tutorial.
+
+You would call this function like so from within trait / struct that implements it:
+
+```
+self.eat_hamburgers(100);
+```
+
+Like this from outside the trait that implements it:
+
+```
+Person::eat_hamburgers(100);
+```
+
+Or like this from outside the struct that implements it:
+
+```
+person.eat_hamburgers(100);
+```
+
+## Creating a structure type object
+
+In Rust creating a structure type object is done very easy:
+
+```
+let given_name = ManagedBuffer::from(b"Bob");
+let given_age = 30u32;
+let person = Person { name: given_name, age: given_age };
+```
+
+If for example you have a variable whose name is matching the structure field's than the syntax can be simplified:
+
+
+```
+let name = ManagedBuffer::from(b"Bob");
+let age = 30u32;
+let person = Person { name, age };
+```
+## Using an indexed mapper
+
+Adding an element to an indexed mapper is done just like adding the an element to the normal one, the only difference being the requirement of the index
+
+
+```
+self.persons(id).set(new_person);
+
+```
+## Reading the value of a mapper
+
+In order to read the value of a `singleValueMapper` we will use the method `get`:
+
+
+```
+let my_person = self.persons(id).get();
+
+```
+## Updating a mapper value
+
+For updating the value of a mapper we can use the `update` method which expects a closure as parameter. You can see the closure as a no name function which we use to access or to process elements in an advaned way within a data type.
+
+```
+let name = ManagedBuffer::from(b"Joe");
+let age = 35u32;
+let new_person = Person { name, age };
+
+self.persons(id).update(|stored_person| *stored_person = new_person );
+```
+Here between `||` we will have the storage type which we can access it under what name we waht, following the operation we want to do to it. Inside the closure we can use variables declared outside. 
```

Comment:
I don't really understand what you're trying to say. I suggestion you rephrase it along these lines: `In the example code above,  between the vertical slashes (`|`), we specified the storage type which we can access using the `new_person` variable...`